test/e2e/v4/plugin_cluster_test.go

```diff
@@ -254,54 +254,53 @@ func Run(kbc *utils.TestContext, hasWebhook, isToUseInstaller, isToUseHelmChart,
 
 	if hasWebhook {
 		By("validating that cert-manager has provisioned the certificate Secret")
-		EventuallyWithOffset(1, func() error {
-			_, err := kbc.Kubectl.Get(
+		EventuallyWithOffset(1, func(g Gomega) {
+			output, err := kbc.Kubectl.Get(
 				true,
 				"secrets", "webhook-server-cert")
-			return err
+			g.Expect(err).ToNot(HaveOccurred(), "webhook-server-cert should exist in the namespace")
+			g.Expect(output).To(ContainSubstring("webhook-server-cert"))
 		}, time.Minute, time.Second).Should(Succeed())
 
 		By("validating that the mutating|validating webhooks have the CA injected")
-		verifyCAInjection := func() error {
+		verifyCAInjection := func(g Gomega) {
 			mwhOutput, err := kbc.Kubectl.Get(
 				false,
 				"mutatingwebhookconfigurations.admissionregistration.k8s.io",
 				fmt.Sprintf("e2e-%s-mutating-webhook-configuration", kbc.TestSuffix),
 				"-o", "go-template={{ range .webhooks }}{{ .clientConfig.caBundle }}{{ end }}")
-			ExpectWithOffset(2, err).NotTo(HaveOccurred())
+			g.ExpectWithOffset(2, err).NotTo(HaveOccurred())
 			// check that ca should be long enough, because there may be a place holder "\n"
-			ExpectWithOffset(2, len(mwhOutput)).To(BeNumerically(">", 10))
+			g.ExpectWithOffset(2, len(mwhOutput)).To(BeNumerically(">", 10))
```

Comment:
The idea is to follow the same approach as the test scaffolds.  
If you look there, you will see that it is quite similar to what we do here.  

For example, we should no longer use `ExpectWithOffset`; instead, we should use the functions and simplify it.

Look at https://github.com/kubernetes-sigs/kubebuilder/issues/4135  

And check the PRs done by @mogsie related to the tests.